src/Uno.UI/UI/Xaml/Media/ImageSource.cs

```diff
@@ -139,13 +148,31 @@ private void InitFromFile(string filePath)
 
 		partial void InitFromResource(Uri uri);
 
-		public static implicit operator ImageSource(string url)
+		partial void CleanupResource();
+
+		public static implicit operator ImageSource(string stringSource)
 		{
-			//This check is done in order to force a null to return if a empty string is passed.
-			return url.IsNullOrWhiteSpace() ? null : new BitmapImage(url);
+			var uri = TryCreateUriFromString(stringSource);
+			return (ImageSource)uri;
 		}
 
-		public static implicit operator ImageSource(Uri uri) => new BitmapImage(uri);
+		public static implicit operator ImageSource(Uri uri)
+		{
+			if (uri is null)
+			{
+				return null;
+			}
+
+			if (uri.LocalPath.EndsWith(".svg", StringComparison.InvariantCultureIgnoreCase) ||
+				uri.LocalPath.EndsWith(".svgz", StringComparison.InvariantCultureIgnoreCase))
```

Comment:
Why `InvariantCultureIgnoreCase`? I think this should be `OrdinalIgnoreCase`.

Same comment applies for all usages of InvariantCulture in SvgImageSource.skia.cs,

Reply:
Indeed, it's a faster comparison technique.